03-transactions.md

```diff
@@ -280,60 +298,73 @@ To redeem the HTLC, the HTLC-success transaction is used as detailed below. This
+  - if `option_anchors` applies:
+    - the `to_local_anchor` output value MUST be subtracted from the `to_local` output.
+    - the `to_remote_anchor` output value MUST be subtracted from the `to_remote` output.
 
 The commitment transaction:
   - if the amount of the commitment transaction `to_local` output would be
-less than `dust_limit_satoshis` set by the transaction owner:
+    less than `dust_limit_satoshis` set by the transaction owner:
     - MUST NOT contain that output.
+    - if `zero_fee_commitments` applies:
+      - MUST add this amount to the `shared_anchor` output.
   - otherwise:
     - MUST be generated as specified in [`to_local` Output](#to_local-output).
   - if the amount of the commitment transaction `to_remote` output would be
-less than `dust_limit_satoshis` set by the transaction owner:
+    less than `dust_limit_satoshis` set by the transaction owner:
     - MUST NOT contain that output.
+    - if `zero_fee_commitments` applies:
+      - MUST add this amount to the `shared_anchor` output.
   - otherwise:
     - MUST be generated as specified in [`to_remote` Output](#to_remote-output).
   - for every offered HTLC:
     - if the HTLC amount minus the HTLC-timeout fee would be less than
-    `dust_limit_satoshis` set by the transaction owner:
+      `dust_limit_satoshis` set by the transaction owner:
       - MUST NOT contain that output.
+      - if `zero_fee_commitments` applies:
+        - MUST add this amount to the `shared_anchor` output.
     - otherwise:
-      - MUST be generated as specified in
-      [Offered HTLC Outputs](#offered-htlc-outputs).
+      - MUST be generated as specified in [Offered HTLC Outputs](#offered-htlc-outputs).
   - for every received HTLC:
     - if the HTLC amount minus the HTLC-success fee would be less than
-    `dust_limit_satoshis` set by the transaction owner:
+      `dust_limit_satoshis` set by the transaction owner:
       - MUST NOT contain that output.
+      - if `zero_fee_commitments` applies:
+        - MUST add this amount to the `shared_anchor` output.
     - otherwise:
-      - MUST be generated as specified in
-      [Received HTLC Outputs](#received-htlc-outputs).
+      - MUST be generated as specified in [Received HTLC Outputs](#received-htlc-outputs).
 
 ## HTLC-Timeout and HTLC-Success Transactions
 
 These HTLC transactions are almost identical, except the HTLC-timeout transaction is timelocked. Both HTLC-timeout/HTLC-success transactions can be spent by a valid penalty transaction.
 
-* version: 2
+* version: 3 when using `zero_fee_commitments`, otherwise 2
 * locktime: `0` for HTLC-success, `cltv_expiry` for HTLC-timeout
 * txin count: 1
    * `txin[0]` outpoint: `txid` of the commitment transaction and `output_index` of the matching HTLC output for the HTLC transaction
-   * `txin[0]` sequence: `0` (set to `1` for `option_anchors`)
+   * `txin[0]` sequence: `1` for `option_anchors`, `0xFFFFFFFD` for `zero_fee_commitments`, `0` otherwise
```

Comment:
What is the difference between using `0` and `0xFFFFFFFD` here?
Neither imposes a bip68 timelock, and both explicitly opt-in to bip125 RBF.

I am just trying to understand the motivation for why the old-style txs were using `0` and why the newer style proposed here uses a different value.

Reply:
I see, so there is no interesting technical difference I missed. Thanks.
I understand the potential privacy aspect, but I don't think it's worth the bikeshedding in this case, so feel free to mark this as resolved :)

Reply:
I *suspect* anyone using TRUC tx will keep this 0 since it's not a meaningful signal, especially post-fullrbf world?

I don't think we have enough data to say exactly.